solr/core/src/java/org/apache/solr/handler/component/SearchComponent.java

```diff
@@ -92,6 +93,9 @@ public void setName(String name) {
     this.name = name;
   }
 
+  public String getJsonKey() { return this.jsonKey; }
+
+  public void setJsonKey(String key) { this.jsonKey = key; }
```

Comment:
WDYT about just having an overridable `getJsonKey()` method that subclasses _can_ override if they want to supply a non-null json key? I don't think there is a need to make this a local variable.

Reply:
SG

Reply:
What does it mean to have a `null` json key?

Reply:
> But it's not super clear to me what a `null` json key here even means.

Yeah agreed, probably should just javadoc the return that `null` means no additional support is needed for json bodies or something like that.

Reply:
Updated the doc comment! Null essentially means "do nothing". Overriding getJsonKey in a SearchComponent means RequestUtil will allow that custom key.